src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -315,12 +316,47 @@ final Type.Declared recordDeclaration(Cursor parent, Cursor recordCursor) {
         ClangSizeOf.with(structOrUnionDecl, recordCursor.type().size() * 8);
         ClangAlignOf.with(structOrUnionDecl, recordCursor.type().align() * 8);
         if (recordCursor.isAnonymousStruct()) {
-            AnonymousStruct.with(structOrUnionDecl);
+            AnonymousStruct.with(structOrUnionDecl, offsetOfAnonymousRecord(parent, recordCursor, recordCursor));
         }
 
         return Type.declared(structOrUnionDecl);
     }
 
+    /*
+     * For the first named field that is nested somewhere inside anonRecord, we get the offset
+     * to outermostParent and anonRecord itself. Subtracting the latter from the former
+     * then gives us the offset of the anonRecord within outermostParent.
+     *
+     * Deals with cases like this too:
+     *
+     * struct Foo {
+     *     char c; // offset = 0
+     *     struct <anon1> { // offset = 96 - 64 = 32
+     *         int: 32;
+     *         struct <anon2> { // offset = 96 - 32 = 64
+     *             int: 32;
+     *             int x; // offset(Foo) = 96, offset(anon2) = 32, offset(anon1) = 64
+     *         };
+     *     };
+     * };
+     */
+    private static OptionalLong offsetOfAnonymousRecord(Cursor outermostParent, Cursor anonRecord, Cursor record) {
```

Comment:
What about DeclarationImpl::recordMemberOffset ? Is that still used? UnsupportedTypeFilter and StructBuilder depend on that for skipping - is is still the case that the first offset might be missing?

Reply:
yes, it's still used. The offset of an anon struct might be missing if it doesn't have any named fields nested inside. In that case, the offset is OptionalLong.empty(), and the UnsupportedFilter adds a Skip to the record.

Reply:
Right... I could simplify the check in UnsupportedFilter to look at the off in the attribute directly, instead of going through `recordMemberOffset`. I think in that case we can also simplify `recordMemberOffset` to return a `long` unconditionally.

Reply:
Addressed in: https://github.com/openjdk/jextract/pull/159/commits/cd00d68d0f0e08912ed358211a663a0fe6274ccb

`recordMemberOffset` is then just used when computing the layout string, so I've move the method to StructBuilder as well.